mz_bokeh_package/utilities/current_user.py

```diff
@@ -0,0 +1,87 @@
+import os
+from bokeh.io import curdoc
+
+from mz_bokeh_package.utilities.environment import Environment
+from mz_bokeh_package.utilities.graphql_api import MZGraphQLClient
+
+
+class CurrentUser:
+    """
+    Class with static methods for getting information about the current user
+    """
+
+    _users = {}
+
+    @classmethod
+    def get_user_info(cls, api_key: str | None) -> dict:
+        """Retrieves user information for the current session or API key, caching the information for future use.
+
+        Args:
+            api_key: unique user api key.
+
+        Returns:
+            dictionary of user "id" and "name".
+
+        Raises:
+             a ValueError if neither a session nor an API key is provided.
+        """
+
+        session_id = curdoc().session_context.id if curdoc().session_context else None
+        if session_id and session_id in CurrentUser._users:
+            return CurrentUser._users[session_id]
+
+        api_key = api_key or CurrentUser.get_api_key()
+        if api_key:
+            user_info = MZGraphQLClient().get_user(api_key)
+            if session_id:
+                cls._cache_user_info(session_id, user_info)
+            return user_info
+        else:
+            raise ValueError("an api_key is required in order to fetch the user info.")
+
+    @classmethod
+    def get_user_id(cls) -> str | None:
+        """get the user_id of the current user
+
+        Returns:
+            the user_id of the current user
+        """
+
+        return cls.get_user_info(CurrentUser.get_api_key()).get("id")
```

Comment:
Is there a case where this method would return `None`? I don't think so. Reading the docstring of `get_user_info`, I see that either `get_user_info` returns a dictionary that has the key `id` or it raises an error. So I think this method always returns `str` and that it should be:
```
return cls.get_user_info(CurrentUser.get_api_key())["id"]
```
If you want to make it slightly more readable, you can go with:
```
user_info = cls.get_user_info(CurrentUser.get_api_key())
return user_info["id"]
```
Same comment for `get_user_name`.